ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -893,9 +893,18 @@ - (void)showViewController:(ORKStepViewController *)viewController goForward:(BO
         }
     }
     
-    if (step.identifier && ![_managedStepIdentifiers.lastObject isEqualToString:step.identifier]) {
-        [_managedStepIdentifiers addObject:step.identifier];
+    NSString *stepIdentifier = step.identifier;
+    if (stepIdentifier) {
+        NSUInteger indexOfStep = [_managedStepIdentifiers indexOfObject:stepIdentifier];
+        if (indexOfStep == NSNotFound) {
+            [_managedStepIdentifiers addObject:stepIdentifier];
+        } else {
+            // If the step being shown has been visited before, remove all the managed step identifiers visited after it
+            NSUInteger numberManagedOfStepIdentifiers = _managedStepIdentifiers.count;
+            [_managedStepIdentifiers removeObjectsInRange:NSMakeRange(indexOfStep + 1, numberManagedOfStepIdentifiers - (indexOfStep + 1))];
```

Comment:
The idea here is that if you loop in a *navigable ordered task*, you don't want to keep duplicate managed step identifiers.

Before this change, managed steps identifiers were being duplicated if you followed a loop, so managed results were showing duplicate entries.

The result of this is that once you take a forward navigation step jump that goes to a previously visited step, all the history after that step is removed (and backwards navigation only takes into account the steps that made you arrive to that step in the first place).

That way, we avoid potentially redundant and very large managed results (and backwards navigation steps).

Navigable ordered tasks with no loops and regular ordered tasks should be unaffected by this change.